tmt/steps/__init__.py

```diff
@@ -531,6 +623,7 @@ def _parse_phases(cls, step):
                     raise tmt.utils.GeneralError(f"Invalid phase '{phase}'.")
             # Store the phase for given step
             try:
+                assert isinstance(phase, int)
```

Comment:
Can you try `cast(Dict[str, int], dict(...))` on line 621? That seems like the last place fooling mypy, all other should lead to the conclusion that `phase` is an integer. This `assert` shouldn't be necessary as long as all assignments are understandable to mypy.

Reply:
Thanks, it worked